RecoTracker/DisplacedRegionalTracking/plugins/DisplacedRegionSeedingVertexProducer.cc

```diff
@@ -121,9 +126,20 @@ void DisplacedRegionSeedingVertexProducer::produce(edm::StreamID streamID,
 
   // Do clustering.
   while (!distances.empty()) {
```

Comment:
Given you are using `std::remove_if` I think there is another optimization you can use. If you do
```
auto itBegin = distances.begin();
auto itLast = distance.end();
```
Then you can have the while loop be
```
while( itBegin != itLast)
```
and then inside the body of the while, use `itLast` instead of `distances.end()` and then change the `std::remove_if` call to be
```
itLast = std::remove_if(itBegin, itLast, distancePred);
```
and then remove the call to `distance.erase(remove_dist, distances.end())` since you no longer care about the full content of the vector.

Reply:
I believe I implemented this..